### DISCUSSION TITLE
[Windows] Pyinstaller onefile structure

### DISCUSSION THREAD

**Discussion by Blank-c** [pyinstaller/pyinstaller#7793] [PyInstaller] [answered]
Can anyone explain me the structure of a `--onefile` output on Windows?

And how can I read the offset of each of them using `PyInstaller.archive.reader`

**Comment by bwoodsend** [comment]
What are you trying to achieve? If you're trying to read a onefile application use `pyi-archive_viewer`. If you're trying to edit one, you'd do better to think of a different way of doing whatever lead you to wanting to edit one.

**Reply by Blank-c** [reply]
No actually I just want to learn about it.

**Reply by rokm** [reply, marked as answer]
There's some description of the CArchive/PKG in documentation: https://pyinstaller.org/en/stable/advanced-topics.html#inspecting-archives

Although, if you want details of the format, you'll have to look at [the code](https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/archive/readers.py).
